base/math.jl

```diff
@@ -841,6 +841,15 @@ function _hypot(x::NTuple{N,<:Number}) where {N}
     end
 end
 
+function _hypot(x::NTuple{N,T}) where {N,T<:IEEEFloat}
+    infT = convert(T, Inf)
+    x = abs.(x) # doesn't change result but enables computational shortcuts
+    any(==(infT), x) && return infT # return Inf even if an argument is NaN
+    maxabs = reinterpret(T, maximum(z -> reinterpret(Signed, z), x)) # for abs(::IEEEFloat) values, a ::BitInteger cast does not change the result
```

Comment:
you should `&` this with `!Base.significand_mask(T)` which will increase accuracy and let you turn the division into a multiplication.

Reply:
I was worried that this would cause a problem in cases where `sqrt(x^2) != abs(x)` (e.g., for `hypot(x,0,0)`). I've since tried every Float32 between 1f0 and 2f0 (WLOG: all values) and billions of Float64s and this never happens, so it appears this won't cause incorrectly rounded calls with a lone nonzero.

See further remarks below.

Reply:
this looks great!